pilot/pkg/serviceregistry/external/servicediscovery.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package external
+
+import (
+	networking "istio.io/api/networking/v1alpha3"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/log"
+)
+
+// Controller communicates with Consul and monitors for changes
+type externalDiscovery struct {
+	config model.IstioConfigStore
```

Comment:
`config` -> `store` so you have less name conflicts.
Stuff like `configs := c.config.ExternalServices()` is confusing.